src/functionality.js

```diff
@@ -0,0 +1,179 @@
+    });
+    moreButton.textContent = '•••';
+    moreButton.classList.add('more-button');
+    moreButton.addEventListener('click', () => {
+      const menu = document.createElement('div');
+      menu.classList.add('menu');
+
+      const buttonsContainer = document.createElement('div');
+      buttonsContainer.style.display = 'flex';
+      buttonsContainer.style.flexDirection = 'column';
+
+      buttonsContainer.appendChild(editButton);
+      buttonsContainer.appendChild(deleteButton);
+
+      menu.appendChild(buttonsContainer);
+
+      listItem.appendChild(menu);
+
+      editButton.style.display = 'block';
+      deleteButton.style.display = 'block';
+      editButton.style.width = '40px';
+      editButton.style.textDecoration = 'none';
+      deleteButton.style.width = '55px';
+      moreButton.style.display = 'none';
+    });
+
+    moreButton.style.float = 'right';
+    moreButton.style.marginRight = '10px';
+    editButton.style.marginTop = '-20px';
+    editButton.style.marginLeft = '90%';
+    deleteButton.style.float = 'right';
+    deleteButton.style.marginLeft = '88%';
+    moreButton.style.border = 'none';
+    moreButton.style.backgroundColor = 'white';
+    listItem.appendChild(moreButton);
+
+    listItem.dataset.index = tasks.length - 1;
+    todoList.appendChild(listItem);
+
+    const hr = document.createElement('hr');
+    todoList.appendChild(hr);
+
+    deleteButton.addEventListener('click', () => {
+      todoList.removeChild(hr);
+    });
+
+    input.value = '';
+    errorMessage.textContent = '';
+  } else {
+    errorMessage.textContent = 'Task cannot be empty!!';
+  }
+  setTimeout(() => {
+    errorMessage.textContent = '';
+  }, 2000);
+};
+
+const input = document.getElementById('type-todo');
+input.addEventListener('keydown', (event) => {
+  if (event.key === 'Enter') {
+    event.preventDefault();
+    addTask();
+  }
+});
+
+const clearCompletedTasks = () => {
+  const todoList = document.getElementById('todo-list');
+  const tasks = todoList.getElementsByTagName('li');
+
+  // Loop through all the tasks in the todo list
+  for (let i = tasks.length - 1; i >= 0; i -= 1) {
+    const task = tasks[i];
+
+    // If the task is completed (i.e., its checkbox is checked)
+    if (task.querySelector("input[type='checkbox']").checked) {
+      // Remove the task and the hr element after it
+      const hr = task.nextElementSibling;
+      todoList.removeChild(task);
+      if (hr && hr.tagName === 'HR') {
+        todoList.removeChild(hr);
+      }
+    }
+  }
+  saveTasksToLocalStorage();
+};
+
+const clearButton = document.getElementById('submit');
+clearButton.addEventListener('click', clearCompletedTasks);
+
+function loadTasksFromLocalStorage() {
+  const storedTasks = JSON.parse(localStorage.getItem('tasks')) || [];
+  tasks.length = 0;
+  tasks.push(...storedTasks);
+  renderTodoList();
+}
+loadTasksFromLocalStorage();
+
+clearButton.addEventListener('click', () => {
+  localStorage.clear();
+});
```

Comment:
- [ ] Please as suggested by the previous reviewer, the `Clear All Completed` button only removes data from the screen and not from the local storage. The data should not be persistent but removed from the local storage when the button is clicked. 😄 
- [ ] Please this is a [link to a loom video](https://www.loom.com/share/e365b4fd43f2434097e2b649d5376ad0) showing the changes required.